pretix_eth/network/networks.py

```diff
@@ -12,20 +39,97 @@ def __init__(self, identifier):
         self.eth_currency_choice = ((eth_currency, _(eth_currency)),)
         self.dai_currency_choice = ((dai_currency, _(dai_currency)),)
 
-    def payment_instructions():
-        raise NotImplementedError("This method has not been implemented for the network")
+    def payment_instructions(
+        self, wallet_address, payment_amount, amount_in_ether_or_dai, currency_type
+    ):
+        """
+        Instructions on how to pay in this network
 
+        :param wallet_address: address to pay to
+        :param payment_amount: amount to pay (in wei)
+        :param amount_in_ether_or_dai: amount to pay (from_wei(payment_amount))
+        :param currency_type: ETH or DAI?
+        :returns dictionary with relevant information like 
+                'erc_681_url', 'uniswap_url', 'web3modal_url', 
+                'amount_manual' (amount + 'ETH/DAI'), 'wallet_address'
+        """
+        
+        raise NotImplementedError("This method has not yet been implemented for the network")
 
 class Rinkeby(INetwork):
     """ Implementation for Rinkeby Testnet """
+
+    CHAIN_ID = 4
+    DAI_ADDRESS = '0xc7AD46e0b8a400Bb3C915120d284AafbA8fc4735'
+
     def __init__(self):
         super(Rinkeby, self).__init__("Rinkeby")
```

Comment:
Just another constructor code-style idea here: why don't you use the `identifier` as a class variable as well, since you are assigning it in the `INetwork` constructor on `self` anyways. And then you can just remove this constructor completely. Of course update the base class constructor not to accept the single argument. It seems bit magical to me, that the base class has different `__init__`'s args than the main classes. Not saying it is wrong - just my opinion.